### PR TITLE
DPL Analysis: add an error message on unbound index access

### PR DESCRIPTION
**Pull request by aalkin**

### CODE REVIEW

**Review by jgrosseo — DISMISSED** [review]

File: Framework/Core/include/Framework/ASoA.h [hunk 1]
```diff
@@ -1664,7 +1668,9 @@ constexpr auto is_binding_compatible_v()
     template <typename T>                                                                                                         \
     auto _Getter_##_as() const                                                                                                    \
     {                                                                                                                             \
-      assert(mBinding != nullptr);                                                                                                \
+      if (O2_BUILTIN_UNLIKELY(mBinding == nullptr)) {                                                                             \
+        throw o2::framework::runtime_error("Index to " #_Table_ " is not bound! Did you subscribe to the table?");                \
+      }                                                                                                                           \
       if (O2_BUILTIN_UNLIKELY(!has_##_Getter_())) {                                                                               \
```
Comment (ktf) [inline Framework/Core/include/Framework/ASoA.h]: We should probably hide all this is a separate, non templated, function.

**Review by ktf — DISMISSED** [review]

**Comment by ktf** [comment]
Looks like there is a conflict, actually...

File: Framework/Core/include/Framework/ASoA.h (hunk 1 above)
Comment (aalkin) [reply Framework/Core/include/Framework/ASoA.h]: @ktf something like this?

File: Framework/Core/src/ASoA.cxx [hunk 2]
```diff
@@ -110,4 +110,11 @@ arrow::ChunkedArray* getIndexFromLabel(arrow::Table* table, const char* label)
   return table->column(index[0]).get();
 }
 
+void checkBinding(void const* binding, const char* tableName)
+{
+  if (O2_BUILTIN_UNLIKELY(binding == nullptr)) {
```
Comment (ktf) [inline Framework/Core/src/ASoA.cxx]: The if statement (and only the `if`) should stay inlined, otherwise you will pay the cost of a function call.

Comment (aalkin) [reply Framework/Core/src/ASoA.cxx]: So only the `throw` should be extracted in a function? Would it have a benefit?

Comment (aalkin) [reply Framework/Core/src/ASoA.cxx]: Should I use `o2::framework::throw_error()` instead?

Comment (ktf) [reply Framework/Core/src/ASoA.cxx]: You benefit the fact that the string / stack unrolling is not embedded in every single instance of that function.

**Review by ktf — APPROVED** [review]

**Pull request merged**